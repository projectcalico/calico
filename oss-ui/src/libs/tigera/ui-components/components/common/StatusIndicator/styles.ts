export default {
    baseStyle: {
        display: 'flex',
        alignItems: 'center',
        '>div': {
            borderRadius: '50%',
            width: '0.5em',
            height: '0.5em',
        },
        gap: 2,
    },
    sizes: {
        xs: {
            fontSize: 'xxs',
            '>div': {
                width: '6px',
                height: '6px',
            },
        },
        sm: {
            fontSize: 'xs',
        },
        md: {
            fontSize: 'sm',
        },
    },
};

export const statusContainerStyles = { alignItems: 'center', w: 10 };
