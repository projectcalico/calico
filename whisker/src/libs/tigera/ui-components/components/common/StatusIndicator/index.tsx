import * as React from 'react';
import type { BoxProps, SystemStyleObject } from '@chakra-ui/react';
import { Box, Flex, Text, useStyleConfig } from '@chakra-ui/react';

interface StatusIndicatorProps extends BoxProps {
    size?: string;
    color: string;
    sx?: SystemStyleObject;
    label: string | number;
    labelSx?: SystemStyleObject;
}

const StatusIndicator: React.FC<StatusIndicatorProps> = ({
    color,
    label,
    labelSx,
    ...rest
}) => {
    const indicatorStyles = useStyleConfig('StatusIndicator', rest);

    return (
        <Flex __css={indicatorStyles} {...rest}>
            <Box bgColor={color}></Box>

            <Text sx={labelSx}>{label}</Text>
        </Flex>
    );
};

export default StatusIndicator;
