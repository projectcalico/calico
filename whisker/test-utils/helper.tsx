import React, { ReactElement } from 'react';
import { render, renderHook, RenderOptions } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '../src/theme';
import { MemoryRouter } from 'react-router';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const queryClientProvider = ({ children }: any) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

export const renderHookWithQueryClient = (hook: () => unknown) =>
    renderHook(hook, { wrapper: queryClientProvider });

const Providers = ({ children }: { children: React.ReactNode }) => {
    return <ChakraProvider theme={theme}>{children}</ChakraProvider>;
};

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options });

export const renderWithRouter = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'> & {
        routes: string[];
    },
) =>
    customRender(
        <MemoryRouter initialEntries={options?.routes}>{ui}</MemoryRouter>,
        options,
    );

export * from '@testing-library/react';

export { customRender as render };
