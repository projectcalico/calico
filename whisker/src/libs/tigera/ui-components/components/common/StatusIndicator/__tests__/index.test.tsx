import { render } from '@/test-utils/helper';
import StatusIndicator from '../index';

describe('<StatusIndicator/>', () => {
    it('it renders', () => {
        const { asFragment } = render(
            <StatusIndicator color={'#f00'} label='foo' />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
