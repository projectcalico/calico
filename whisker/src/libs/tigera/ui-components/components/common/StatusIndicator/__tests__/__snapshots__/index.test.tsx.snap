// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusIndicator/> it renders 1`] = `
<DocumentFragment>
  <div
    class="css-1tljkm1"
  >
    <div
      class="css-fw3qcu"
    />
    <p
      class="chakra-text css-0"
    >
      foo
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
